client/header/activity-panel/panels/inbox/index.js

```diff
@@ -66,32 +47,61 @@ class InboxPanel extends Component {
 		);
 	}
 
-	renderNotes() {
-		const { lastRead, notes } = this.props;
+	renderNotes( hasNotes ) {
+		const { isUndoRequesting, lastRead, notes } = this.props;
 
-		const validNotes = filter( notes, ( note ) => {
-			const { is_deleted: isDeleted } = note;
-			const noteActive = has( note, 'is_deleted' ) ? ! isDeleted : true;
-			return noteActive;
-		} );
+		if ( isUndoRequesting === 'undo-dismiss-all' ) {
+			return;
+		}
 
-		if ( validNotes.length === 0 ) {
+		if ( ! hasNotes ) {
 			return this.renderEmptyCard();
 		}
 
 		const notesArray = Object.keys( notes ).map( ( key ) => notes[ key ] );
 
-		return notesArray.map( ( note ) => (
-			<InboxNoteCard
-				key={ note.id }
-				note={ note }
-				lastRead={ lastRead }
-			/>
-		) );
+		return notesArray.map( ( note ) => {
+			const showPlaceholder = isUndoRequesting
+				? isUndoRequesting === note.id
+				: false;
+			if ( showPlaceholder ) {
+				return (
+					<InboxNotePlaceholder
+						className={ 'banner message-is-unread' }
+						key={ note.id }
+					/>
+				);
+			}
+			return (
+				<InboxNoteCard
+					key={ note.id }
+					note={ note }
+					lastRead={ lastRead }
+				/>
+			);
+		} );
+	}
+
+	renderNotePlaceholder( isRequesting, isUndoRequesting ) {
+		if ( isRequesting || isUndoRequesting === 'undo-dismiss-all' ) {
+			return (
+				<Section>
+					<InboxNotePlaceholder
+						className={ 'banner message-is-unread' }
```

Comment:
Nitpick: this could just use a double quoted attribute instead braces.

Reply:
Good idea. I modified it in the commit [2351730](https://github.com/woocommerce/woocommerce-admin/pull/4379/commits/2351730652923aca87dd93a071d37cb706e6beed)